plugins/modules/keycloak_authentication_required_actions.py

```diff
@@ -0,0 +1,457 @@
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_authentication_required_actions
+
+short_description: Allows administration of Keycloak authentication required actions
+
+description:
+    - This module can register, update and delete required actions.
+    - It also filters out any duplicate required actions by their alias. The first ocurrence is preserved.
+
+version_added: 7.1.0
+
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: full
+
+options:
+    realm:
+        description:
+            - The name of the realm in which are the authentication required actions.
+        required: true
+        type: str
+    required_actions:
+        elements: dict
+        description:
+            - Authentication required action.
+        suboptions:
+            alias:
+                description:
+                    - Unique name of the required action.
+                required: true
+                type: str
+            config:
+                description:
+                    - Configuration for the required action.
+                type: dict
+            defaultAction:
+                description:
+                    - Indicates, if any new user will have the required action assigned to it.
+                type: bool
+            enabled:
+                description:
+                    - Indicates, if the required action is enabled or not.
+                type: bool
+            name:
+                description:
+                    - Displayed name of the required action. Required for registration.
+                type: str
+            priority:
+                description:
+                    - Priority of the required action.
+                type: int
+            providerId:
+                description:
+                    - Provider ID of the required action. Required for registration.
+                type: str
+        type: list
+    state:
+        choices: [ "absent", "present" ]
+        description:
+            - Control if the realm authentication required actions are going to be registered/updated (V(present)) or deleted (V(absent)).
+        required: true
+        type: str
+
+extends_documentation_fragment:
+    - community.general.keycloak
+    - community.general.attributes
+
+author:
+    - Skrekulko (@Skrekulko)
+'''
+
+EXAMPLES = '''
+- name: Register a new required action.
+  community.general.keycloak_authentication_required_actions:
+  auth_client_id: "admin-cli"
+  auth_keycloak_url: "http://localhost:8080"
+  auth_password: "password"
+  auth_realm: "master"
+  auth_username: "admin"
+  realm: "master"
+  required_action:
+    - alias: "TERMS_AND_CONDITIONS"
+      name: "Terms and conditions"
+      providerId: "TERMS_AND_CONDITIONS"
+      enabled: true
+  state: "present"
```

Comment:
Well, this part must be indented, otherwise ansible won't accept the task:
```suggestion
    auth_client_id: "admin-cli"
    auth_keycloak_url: "http://localhost:8080"
    auth_password: "password"
    auth_realm: "master"
    auth_username: "admin"
    realm: "master"
    required_action:
      - alias: "TERMS_AND_CONDITIONS"
        name: "Terms and conditions"
        providerId: "TERMS_AND_CONDITIONS"
        enabled: true
    state: "present"
```
Same below.